clang/lib/Parse/ParseDecl.cpp

```diff
@@ -7893,7 +7893,7 @@ void Parser::ParseFunctionDeclarator(Declarator &D,
   StartLoc = LParenLoc;
 
   /* TO_UPSTREAM(BoundsSafety) ON */
-  LateParsedAttrList LateParamAttrs(/*PSoon=*/false,
+  LateParsedAttrList LateParamAttrs(/*PSoon=*/true,
```

Comment:
I think we need a more detailed explanation of `PSoon`.
Last time when I made changes in `ParseBracketDeclarator`, I also toggled the `PSoon` argument from false to true there.  In the case I investigated, setting `PSoon` to false results in late processing of those attributes after the entire C++ class had been parsed.  It was incorrect because those attributes belong to member function parameters.  In general, I do not fully understand this flag.

At that time, I also checked all call sites of this constructor.  IIRC, with your change here,  all call sites will be passing `true` to `PSoon`.  If this is true, maybe we should consider removing this flag?

Reply:
To be honest, my understanding is somewhat shallow also. But basically, not parsing and removing the late attributes in this context will lead to an error when parsed later, after finishing parsing of the C++ class. Calling `ParseLexedCAttributeList` requires PSoon to be true because of an assert, so I changed it. Since we actually are parsing directly, I don't see why not.
@rapidsna do you have any context to add about whether we still need PSoon?